src/main/java/dev/slimevr/vr/processor/skeleton/LegTweakBuffer.java

```diff
@@ -0,0 +1,484 @@
+	private Vector3f rightFootPosition = placeHolderVec;
+	private Vector3f leftKneePosition = placeHolderVec;
+	private Vector3f rightKneePosition = placeHolderVec;
+	private Vector3f waistPosition = placeHolderVec;
+	private Quaternion leftFootRotation = placeHolderQuat;
+	private Quaternion rightFootRotation = placeHolderQuat;
+
+	private Vector3f leftFootPositionCorrected = placeHolderVec;
+	private Vector3f rightFootPositionCorrected = placeHolderVec;
+	private Vector3f leftKneePositionCorrected = placeHolderVec;
+	private Vector3f rightKneePositionCorrected = placeHolderVec;
+	private Vector3f waistPositionCorrected = placeHolderVec;
+
+	// velocities
+	private Vector3f leftFootVelocity = placeHolderVec;
+	private float leftFootVelocityMagnitude = 0;
+	private Vector3f rightFootVelocity = placeHolderVec;
+	private float rightFootVelocityMagnitude = 0;
+	private float leftFootAngleDiff = 0;
+	private float rightFootAngleDiff = 0;
+
+	// acceleration
+	private Vector3f leftFootAcceleration = placeHolderVec;
+	private float leftFootAccelerationMagnitude = 0;
+	private Vector3f rightFootAcceleration = placeHolderVec;
+	private float rightFootAccelerationMagnitude = 0;
+
+	// other data
+	private long timeOfFrame = System.nanoTime();
+	private LegTweakBuffer parent = null; // frame before this one
+	private int frameNumber = 0; // higher number is older frame
+	private int detectionMode = ANKLE_ACCEL; // detection mode
+	private boolean accelerationAboveThresholdLeft = true;
+	private boolean accelerationAboveThresholdRight = true;
+	private float leftFloorLevel;
+	private float rightFloorLevel;
+
+	// hyperparameters
+	public static final float SKATING_DISTANCE_CUTOFF = 0.225f;
+	private static final float SKATING_VELOCITY_THRESHOLD = 4.25f;
+	private static final float SKATING_ACCELERATION_THRESHOLD = 1.15f;
+	private static final float SKATING_ROTVELOCITY_THRESHOLD = 2.8f;
+	private static final float SKATING_LOCK_ENGAGE_PERCENT = 0.9f;
+	private static final float FLOOR_DISTANCE_CUTOFF = 0.1f;
+	private static final float SIX_TRACKER_TOLLERANCE = 0.10f;
+
+	private static final float PARAM_SCALAR_MAX = 1.75f;
+	private static final float MAX_SCALAR_ACCEL = 0.3f;
+	private static final float MIN_SCALAR_ACCEL = 1.25f;
+
+	private float leftFootSensitivity = 1.0f;
+	private float rightFootSensitivity = 1.0f;
+
+
+	private static final float SKATING_CUTOFF_ENGAGE = SKATING_DISTANCE_CUTOFF
+		* SKATING_LOCK_ENGAGE_PERCENT;
+	private static final float SKATING_VELOCITY_CUTOFF_ENGAGE = SKATING_VELOCITY_THRESHOLD
+		* SKATING_LOCK_ENGAGE_PERCENT;
+	private static final float SKATING_ACCELERATION_CUTOFF_ENGAGE = SKATING_ACCELERATION_THRESHOLD
+		* SKATING_LOCK_ENGAGE_PERCENT;
+	private static final float SKATING_ROTATIONAL_VELOCITY_CUTOFF_ENGAGE = SKATING_ROTVELOCITY_THRESHOLD
+		* SKATING_LOCK_ENGAGE_PERCENT;
+
+	// getters and setters
+	public Vector3f getLeftFootPosition() {
+		return leftFootPosition.clone();
+	}
```

Comment:
Could we have getters that follow the existing convention? Something like this:
```java
public boolean getLeftFootPosition(Vector3f vec) {
	return vec.set(leftFootPosition);
}
```

Reply:
Yes